storage/jellyfish-merkle/src/node_type/mod.rs

```diff
@@ -136,26 +136,65 @@ impl NodeKey {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum NodeType {
+    Leaf,
+    /// A internal node that haven't been finished the leaf count migration, i.e. None or not all
+    /// of the children leaf counts are known.
+    InternalLegacy,
```

Comment:
To clarify, this is a node where the counts of at least one child is unknown, right?